File: src/agents/Kotlin/probe.py
from src.agents.base import BaseActivePersona
import logging
import time
from pathlib import Path

logger = logging.getLogger(__name__)

class ProbePersona(BaseActivePersona):
    """
    Core: PhD in Software Forensics & Android Diagnostics 🔍
    Monitorado por Metric: Injeção de Telemetria de Auditoria.
    """
    
    def __init__(self, project_root):
        super().__init__(project_root)
        self.name, self.emoji, self.role, self.stack = "Probe", "🔍", "PhD Diagnostics Engineer (Kotlin)", "Kotlin"

    def perform_audit(self) -> list:
        """Auditoria com telemetria forense JVM integrada."""
        start_time = time.time()
        logger.info(f"[{self.name}] Analisando Integridade Android...")
        
        # Sintaxe linear
        rules = [
            {'regex': r"catch\s*\(.*?:\s*Exception\)\s*\{\s*\}", 'issue': 'Silenciamento Crítico: Erro ignorado na JVM.', 'severity': 'critical'},
            {'regex': r"Log\.d\(", 'issue': 'Vazamento de Contexto: Log.d em produção.', 'severity': 'medium'}
        ]
        
        results = self.find_patterns(('.kt', '.kts'), rules)
        
        duration = time.time() - start_time
        logger.info(f"🔍 [{self.name}] Auditoria finalizada em {duration:.4f}s. Anomalias: {len(results)}")
        return results

    def _reason_about_objective(self, objective, file, content):
        if "catch" in content and "Exception" in content and "{}" in content:
            return f"Instabilidade Oculta: O objetivo '{objective}' exige resiliência. Em '{file}', o silenciamento de falhas JVM impede a auto-correção da 'Orquestração de Inteligência Artificial'."
        return None

    def get_system_prompt(self):
        return f"Você é o Dr. {self.name}, mestre em diagnósticos Kotlin."

File: src_local/agents/Python/Audit/probe.ts
import { BaseActivePersona } from "../../base.ts";
import type { AuditFinding, AuditRule, StrategicFinding } from "../../base.ts";
import type { ProjectContext } from "../../../core/types.ts";

/**
 * 🔬 Dr. Probe — PhD in Python Error Resilience & Exception Handling
 */
export class ProbePersona extends BaseActivePersona {
    constructor(projectRoot: string | undefined = undefined) {
        super(projectRoot);
        this.name = "Probe";
        this.emoji = "🔬";
        this.role = "PhD Resilience Engineer";
        this.phd_identity = "Python Error Resilience & Exception Handling";
        this.stack = "Python";
    }

    override async execute(context: ProjectContext): Promise<AuditFinding[]> {
        this.setContext(context);
        const findings = await this.performAudit();
        return findings;
    }

    override getAuditRules(): { extensions: string[]; rules: AuditRule[] } {
        return {
            extensions: [".py"],
            rules: [
                { regex: /except\s*:\s*pass/, issue: "Cegueira Python: Catch vazio (pass) detectado.", severity: "critical" }
            ]
        };
    }

    public audit(): any[] { return []; }
    public Branding(): string { return `${this.emoji} ${this.name}`; }
    public Analysis(): string { return "Resilience Analysis Complete"; }
    public test(): boolean {
        this.Branding();
        this.Analysis();
        this.audit();
        return true;
    }

    override reasonAboutObjective(objective: string, file: string, content: string | Promise<string | null>): StrategicFinding | string | null {
        return {
            file, severity: "INFO",
            issue: `PhD Probe (Python): Analisando integridade de erros para ${objective}.`,
            context: "analyzing resilience"
        };
    }

    override selfDiagnostic(): any {
        return { status: "Soberano", score: 100, issues: [], branding: this.Branding() };
    }

    override getSystemPrompt(): string {
        return `Você é o Dr. ${this.name}, PhD em resiliência Python. Status: ${this.Analysis()}`;
    }
}
